### ISSUE TITLE
[CHORE packages] enhance `packages-for-commit` for shareable tarballs

### ISSUE THREAD

**Issue by runspired** [emberjs/data#6119]
Currently the tarballs produced by `node ./bin/packages-for-commit.js` are absolute paths, making it difficult to check those tarballs into CI or share them with someone for a local test run. We should make them use relative paths such that as long as they are within the same directory as each other they work.

Steps today for testing a given commit for instance

**Generate Tarballs**
```
git clone git@github.com:emberjs/data.git;
cd data;
git checkout <sha>;
yarn install;
node ./bin/packages-for-commit.js
```

Then  in the consuming project's `package.json`

```
"ember-data": "file:${pathToGeneratedEmberDataTarball}"
```

Because the tarballs use absolute paths they effectively only work well on the same machine as the project that generated them. If we make them relative then either project contributors or a PR author needing to test a PR could produce the tarballs easily and then check them in with a project to allow CI to use them.

**Bonus Feature**

A bonus feature is to have our CI publish any builds for PR commits to a registry we own such that someone could download the tarballs for that build for other test runs (probably hosted on azure/heroku using verdaccio)

**Comment by pete-the-pete** [comment]
Small note on this since I missed it, `node ./bin/packages-for-commit.js` needs to be run _before_ running external tests so that the tarballs exist (they aren't created automatically).

**Comment by runspired** [comment]
Right, this ticket is for improving the existing process, the description above is documenting what the process is today with a description of some improvements to make to it.

**Comment by runspired** [comment]
#6128 upgraded our script for being able to generate tarballs for various situations (local file path, remote/hosted file path, custom registry)

However, we should still do more to ensure that folks submitting PRs have a pleasant experience when attempting to test their PR against their application, will update the ticket above with the requirements for that.

**Comment by runspired** [comment]
Resolved by #6128
